### ISSUE TITLE
GameGuru MAX - Request - Swap Object Function in Editor

### ISSUE THREAD

**Issue by MonkeyFrogStudio** [TheGameCreators/GameGuruRepo#5502]
I think it would be great if there were a simple way, in the editor, to swap or change one current object for another ... maybe by selecting the object, accessing a right-click menu, and selecting "Swap Object", which would then open the object library for you to select a new object to replace the selected one.

Why would we need this?

A few reasons:

1 - I'm creating a DLC that has several versions of the same object - clean, dirty, and bloody. I might place a clean version of an object and, later, decide I'd rather use the dirty version. If I have something already precisely placed, then I don't want to have to delete the current object, bring in the new object, and precisely place it ... not if I could just click-n-replace. ;)

2 - This could help with 'grayboxing'. Grayboxing is when the level designer uses temp geometry (normally bland, gray simple shapes) so they can test things like the flow of their game, to see if players can get stuck in places, etc. At some point, the simple geometry needs to get replaced with the complex, final geometry. Currently, we would have to delete each object and replace manually. If we had a means of swapping out objects, this would simplify the process greatly.

Thanks for considering this idea.

**Comment by Necrym59** [comment]
1. I think is a great idea, very handy for different variations using the same model
2. Im not sure about, the object to swap/replace would have to have exactly the same dimensions, xyz origin and bounding etc. in which case the original could have been used initially.

**Comment by MonkeyFrogStudio** [comment]
On #2 - In many cases, yes. But not all. For example, I could use simple boxes to make the walls of a valley, giving me the approximate shape of the 'path' through the valley. Then, after testing the gameplay, I could swap out the simple boxes with complex rock walls that have details, etc. The boxes wouldn't have to be as tall as the final versions, etc. And since the walls need not be exact, swapping would work here.

In another example, I could have a room and use a simple block as a placeholder for a table. More blocks as placeholders for chairs. Then I could swap out the blocks for the table and the chairs, etc. 

Working like this (which is very typical for grayboxing) allows for quickly prototyping a level, testing gameplay, etc., before COMMITTING to final model design/creation.

**Comment by Necrym59** [comment]
Yeah i suppose there is a use for some, ive always found grayboxing pointless nowadays but some may use it

**Comment by MonkeyFrogStudio** [comment]
It's how most games are initially developed in AAA studios, even now.

**Comment by Necrym59** [comment]
Only a couple of game studios ive worked for actualy used the greyboxing technique. Even some of the bigger MMO games ive worked on never used it. Probably depends on the structure of the teams and how they develop as there is no hard fast rule to use it.

**Comment by MonkeyFrogStudio** [comment]
Sigh. Epic Games used it for games like Fortnight and Rocket League. Is that recent and large enough? The technique is also known as 'blockout' and also has other names. It is the primary reason that engines like Unreal, Unity, and others provide a set of adjustable base forms, like cubes, cylinders, and spheres - so that game developers can blockout levels before committing to final design. There are numerous studios, small and large, that use the technique in a variety of ways for a variety of reasons. 

Whatever your personal experiences have been, there are numerous people who use it, numerous videos explaining how and why to use it (many made by the studios and professional artists who use this technique to professionally create levels), and some of us who would like to use it in MAX. Yet it's not practical in MAX as it is. Thus, having something like a 'swap' feature would be a decent start and, as pointed out, it could be used for other reasons.

As stated, the DLC I am developing has clean, dirty, and bloody versions of most objects, including walls and floors. I could create a level that's all clean. Then, I could use the same level, but with the dirty, ruined versions. One level could be the 'before' some event happened and the other after this event (like for flashbacks or some other reason). It would be simple to be able to swap out the models than to have to delete, add in, and position exactly as the other.

**Comment by Necrym59** [comment]
Im sorry I didnt say it wasnt useful, I was just talking from my personal experiences in the industry - its getting impossible to talk to you without you taking a counter view and a lecture in your view of the world - im over it.

**Comment by MonkeyFrogStudio** [comment]
Like your above comment, it didn't add to the actual request. Stating that most studios you've worked with don't use the technique does not help the request or show why it should or shouldn't be considered, etc. This is not a discussion board, so what was the point? And that's why I countered ... to support my request by providing reasons why and how a 'swap' feature could be used in MAX. If you have reasons that support the request, fine. Post that. If you have ways that make the request better. Fine, post that. If you have actual reasons the request is not good, impractical, etc., fine, post that. It's quite simple.
